### PR TITLE
Optimize ZSTDMT_compress() memory usage

### PR DESCRIPTION
**Pull request by Cyan4973**
will no longer allocated temporary buffer
if there is enough space in provided `dstBuffer` to compress directly there.

### CODE REVIEW

File: lib/compress/zstdmt_compress.c [hunk 1]
```diff
@@ -388,14 +391,17 @@ size_t ZSTDMT_compressCCtx(ZSTDMT_CCtx* mtctx,
...
     size_t const chunkTargetSize = (size_t)1 << (params.cParams.windowLog + 2);
-    unsigned const nbChunksMax = (unsigned)(srcSize / chunkTargetSize) + (srcSize < chunkTargetSize) /* min 1 */;
+    unsigned const nbChunksMax = (unsigned)(srcSize / chunkTargetSize) + 1;
     unsigned nbChunks = MIN(nbChunksMax, mtctx->nbThreads);
     size_t const proposedChunkSize = (srcSize + (nbChunks-1)) / nbChunks;
     size_t const avgChunkSize = ((proposedChunkSize & 0x1FFFF) < 0xFFFF) ? proposedChunkSize + 0xFFFF : proposedChunkSize;   /* avoid too small last block */
     size_t remainingSrcSize = srcSize;
     const char* const srcStart = (const char*)src;
-    size_t frameStartPos = 0;
+    unsigned const compressWithinDst = (dstCapacity >= ZSTD_compressBound(srcSize)) ? nbChunks : (unsigned)(dstCapacity / ZSTD_compressBound(avgChunkSize));  /* presumes avgChunkSize >= 256 KB, which should be the case */
```
Comment (terrelln) [inline lib/compress/zstdmt_compress.c]: Why does `avgChunkSize` need to be at least 256 KB?

File: programs/zstd.1.md [hunk 2]
```diff
@@ -290,6 +303,19 @@ The list of available _options_:
...
+    Select the amount of data reloaded from previous job into next one.
+    Reloading more data improves compression ratio, but decreases speed.
+    This parameter is only available if multithreading is enabled.
+
+    The minimum _ovlog_ is 0, and the maximum is 9.
+    0 means "no overlap", hence completely independent jobs.
+    9 means "full overlap", meaning up to `windowSize` is reloaded from previous job.
+    Reducing _ovlog_ by 1 reduces the amount of reload by a factor 2.
+    Default _ovlog_ is 6, which means "reload `windowSize / 8`".
+    Exception : the maximum compression level (22) has a default _ovlog_ of 9.
```
Comment (terrelln) [inline programs/zstd.1.md]: Mention that `overlapSize` is derived from `overlapLog`.

File: lib/compress/zstdmt_compress.c (hunk 1 above)
Comment (Cyan4973) [inline lib/compress/zstdmt_compress.c]: Below that threshold, `ZSTD_compressBound()` does no longer guarantee that `ZSTD_compressBound(A+B) >= ZSTD_compressBound(A) + ZSTD_compressBound(B)`

**Comment by Cyan4973** [comment]
man updated for `overlapSize`

**Pull request merged**